### PR TITLE
[Numpy] Refactor Roberta

### PR DESCRIPTION
**Pull request by ZheyuYe**
## Description ##
Refactor reoberta model following https://github.com/apache/incubator-mxnet/issues/18717 

### Changes ###
- [x] Refactor reoberta and maintain consistency in model structure and external APIS with other pretrained models
- [x] Maintain consistency in the interface of tokenizers
- [x] Store the converted weights sperately for backbone and masked language model
- [x] Revise run_squad.py to support roberta fine-tuning

## Comments ##
@sxjscience  @hymzoque

### CODE REVIEW

**Comment by leezu** [comment]
It's unclear what part of the PR is related to https://github.com/apache/incubator-mxnet/issues/18717 and what part are unrelated changes. I suggest to focus on making the https://github.com/apache/incubator-mxnet/issues/18717 feature available in MXNet instead of making the GluonNLP code more complex to workaround the missing feature. What do you think?

Is there any overlap between the weights "Store the converted weights sperately for backbone and masked language model"?

File: scripts/question_answering/models.py [hunk 1]
```diff
@@ -14,9 +14,11 @@ class ModelForQABasic(HybridBlock):
     another dense layer to map the contextual embeddings to the start scores and end scores.
 
     """
-    def __init__(self, backbone, weight_initializer=None, bias_initializer=None):
+    def __init__(self, backbone, weight_initializer=None, bias_initializer=None,
+                 use_segmentation=True):
```
Comment (sxjscience) [inline scripts/question_answering/models.py:18]: Add more documentations to explain this new flag.

**Comment by ZheyuYe** [comment]
@leezu There is a huge overlap between 'model.param' and 'model-mlm.params' which could be eliminated by https://github.com/apache/incubator-mxnet/pull/18749 by only storing 'model-mlm.params' and load it with `ignore_extra=True` for backbone model. It can be done in a separate PR for roberta as well as other pretrain models that use sperate parameters if this solution is reasonable for you. 

In addition, this PR also solves the problem that the current roberta model does not handle MLM task properly. That is, the MLM model takes the same input (`input_ids, valid_length`) as backbone without marking which positions are masked as https://github.com/dmlc/gluon-nlp/blob/e78a24eeb124ff9efb99b1aee7ab1ee9181c9e0f/src/gluonnlp/models/bert.py#L476 and [offical implementation](https://github.com/pytorch/fairseq/blob/master/fairseq/models/roberta/model.py#L249-L250 )

**Comment by leezu** [comment]
Thanks @ZheyuYe. Yes, I recommend to adding features in the MXNet side instead of adopting a problematic workaround for GluonNLP

**Comment by ZheyuYe** [comment]
> Thanks @ZheyuYe. Yes, I recommend to adding features in the MXNet side instead of adopting a problematic workaround for GluonNLP

Alright, I am going to combine backbone and it's mlm models together in this PR once
https://github.com/apache/incubator-mxnet/pull/18749 merged

File: scripts/question_answering/README.md [hunk 2]
```diff
@@ -128,8 +152,9 @@ For BERT and ELECTRA model, the results on SQuAD1.1 and SQuAD2.0 are given as fo
 |ELECTRA small             | 85.42/78.95   | 74.44/71.86  |        
 |ELECTRA base              | 92.63/87.34   | 86.34/83.62  |
 |ELECTRA large             | 94.95/89.94   | 90.59/88.13  |
+|RoBERTa larger            | 88.86/94.58   | 85.93/89.01  |
```
Comment (sxjscience) [inline scripts/question_answering/README.md:155]: Should be 89.01/85.93

File: scripts/question_answering/README.md [hunk 3]
```diff
@@ -138,5 +163,7 @@ For reference, we have also included the results of Google's original version
 |Google ELECTRA base       |     - /75.8    |     - /70.1   |
 |Google ELECTRA base       |     - /86.8    |     - /83.7   |
 |Google ELECTRA large      |     - /89.7    |     - /88.1   |
+|Fairseq RoBERTa large     |   88.9/94.6    |	86.5/89.4   |
```
Comment (sxjscience) [inline scripts/question_answering/README.md:166]: Should be 94.6/88.9, 89.4/86.5. Also add the documentation about F1 + EM.

File: src/gluonnlp/models/roberta.py [hunk 4]
```diff
@@ -173,32 +176,29 @@ def __init__(self,
...
         self.use_pooler = use_pooler
-        self.use_mlm = use_mlm
-        self.untie_weight = untie_weight
+        self.classifier_activation = classifier_activation
         self.encoder_normalize_before = encoder_normalize_before
-        self.return_all_hiddens = return_all_hiddens
-        self.tokens_embed = nn.Embedding(
+        self.weight_initializer = weight_initializer
+        self.embed_initializer = embed_initializer
+        self.bias_initializer = bias_initializer
```
Comment (sxjscience) [inline src/gluonnlp/models/roberta.py:184]: There is no need to store it to `self`.

File: scripts/question_answering/README.md (hunk 2 above)
Comment (sxjscience) [reply scripts/question_answering/README.md:155]: Also should be RoBERTa large.

**Comment by sxjscience** [comment]
@leezu Why is it blocked by https://github.com/apache/incubator-mxnet/pull/18749 ?
